src/progress.jl

```diff
@@ -0,0 +1,110 @@
+import Base: done
+
+export ProgressBar, progress!, msg!, done, right_text!, @progress
+
+type ProgressBar
+  leftText::String
+  rightText::String
+  id::String
+  progress::Float64
+  msg::String
+  determinate::Bool
```

Comment:
Given that `progress` and `determinate` are exclusive anyway, could we use a special value like `NaN` for indeterminate bars?